src/ATen/native/xpu/sycl/ForeachCopyKernels.cpp

```diff
@@ -17,23 +76,49 @@ struct Identity {
 void foreach_copy_list_kernel_(TensorList self, TensorList src) {
   std::vector<std::vector<at::Tensor>> tensor_lists{src.vec(), self.vec()};
 
-  AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND3(
+  AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND7(
       at::ScalarType::Half,
       at::ScalarType::BFloat16,
       at::ScalarType::Bool,
+      ScalarType::Float8_e4m3fn,
+      ScalarType::Float8_e4m3fnuz,
+      ScalarType::Float8_e5m2,
+      ScalarType::Float8_e5m2fnuz,
       self[0].scalar_type(),
       "foreach_tensor_copy",
       [&]() {
         using opmath_t = at::opmath_type<scalar_t>;
-        multi_tensor_apply<2>(
-            tensor_lists,
-            UnaryOpFunctor<
-                scalar_t,
-                /* depth */ 2,
-                /* r_args_depth */ 1,
-                /* res_arg_index */ 1>(),
-            Identity<opmath_t>());
+        AT_DISPATCH_SOURCE_TYPES(
+            src[0].scalar_type(), "foreach_tensor_copy", [&] {
+              multi_tensor_apply<2>(
+                  tensor_lists,
+                  UnaryOpFunctor<
+                      scalar_t,
+                      /* depth */ 2,
+                      /* r_args_depth */ 1,
+                      /* res_arg_index */ 1>(),
+                  Identity<opmath_t>());
+            });
       });
+
+  // AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND3(
```

Comment:
what's the reason to have this commented out? Either remove or add in-source comment explaining why this might be needed.